src/Workspaces/Core/Desktop/Workspace/FileTextLoader.cs

```diff
@@ -101,6 +101,34 @@ protected virtual SourceText CreateText(Stream stream, Workspace workspace)
             return textAndVersion;
         }
 
+        internal override TextAndVersion LoadTextAndVersionSynchronously(Workspace workspace, DocumentId documentId, CancellationToken cancellationToken)
+        {
+            DateTime prevLastWriteTime = FileUtilities.GetFileTimeStamp(_path);
```

Comment:
While you're in this code.  Can you fix `GetFileTimestamp`?  

``` csharp
            catch (Exception e)
            {
                throw new IOException(e.Message);
            }
```

That needs to put `e` as the inner exception (second parameter).